Rename Publisher to Reporting Org in UI

simon-20 opened the pull request:
This renames [Pp]ublisher(s) to [Rr]eporting [Oo]rg across the UI of the Dashboard, without altering any code (other than code containing site text) or making any changes to URLs.

Where 'Reporting Org' appears in a heading I've capitalised both words.

Where 'Reporting org' appears at the start of a sentence, I've decapitalised 'org'.

I've also adjusted the title and headers on the 'Reporting Org' errors page, because with 'Reporting Org' now being used instead of 'Publisher', it was confusing to have an errors page entitled 'Reporting Orgs'.

Bjwebb commented:
A reminder (mostly to myself): we should wait for https://github.com/IATI/IATI-Dashboard/pull/738 to be deployed before we merge this, so that we can have this on the dev site to review for a bit.

Bjwebb commented on iati_dashboard/templates/summary_stats.html:
```diff
@@ -19,7 +19,7 @@ <h2 class="dashboard-panel-heading__title float-start">Summary Statistics</h2>
 
         <div class="dashboard-panel-body">
-            <p>This table assesses IATI publishers by scoring three dimensions: Timeliness, Forward-looking and Comprehensiveness. The methodology is explained below the table and in the related Publisher Statistics pages. In summary:</p>
+            <p>This table assesses IATI reporting orgs by scoring three dimensions: Timeliness, Forward-looking and Comprehensiveness. The methodology is explained below the table and in the related Reporting Org Statistics pages. In summary:</p>
```
I think this should read "Publishing Statistics pages" (ie. it was wrong before)

simon-20 replied:
Fixed

Bjwebb reviewed (changes requested):
Can we link to https://github.com/IATI/IATI-Dashboard/issues/729 in the commit message?

simon-20 committed:
feat: rename Publisher to Reporting Org in UI

This makes changes to the UI (not including public-facing URLs)
that renames Publisher -> Reporting Org and so resolves
https://github.com/IATI/IATI-Dashboard/issues/729.
